src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
Does this even work? We tried using extends before and ran into issues that forced double casting to make it work, hence why it was simplified